src/gmpy2_convert.c

```diff
@@ -166,45 +166,61 @@ GMPy_RemoveUnderscoreASCII(PyObject *s)
  */
 
 static int
-mpz_set_PyStr(mpz_ptr z, PyObject *s, int base)
+mpz_set_PyStr(mpz_t z, PyObject *s, int base)
 {
-    char *cp;
-    Py_ssize_t len;
-    PyObject *ascii_str = ascii_str = GMPy_RemoveUnderscoreASCII(s);
+    char *cp, negative = 0;
+    PyObject *ascii_str;
+
+    ascii_str = GMPy_RemoveUnderscoreASCII(s);
 
     if (!ascii_str) return -1;
 
-    len = PyBytes_Size(ascii_str);
     cp = PyBytes_AsString(ascii_str);
 
+    while (*cp == ' ') cp++;
```

Comment:
I wonder if we shouldn't modify GMPy_RemoveUnderscoreASCII() helper to strip white spaces too.

Reply:
I think removing underscore and spaces in one pass makes sense.

Reply:
I had some doubts about the mpc case...

Ok, I'll do this.  (Not sure how to rename GMPy_RemoveUnderscoreASCII() helper...)